internals/secrethub/inject.go

```diff
@@ -78,12 +88,48 @@ func (cmd *InjectCommand) Run() error {
 		return errio.Error(err)
 	}
 
-	varTemplate, err := tpl.NewV1Parser().Parse(string(raw))
+	templateVars := make(map[string]string)
```

Comment:
I feel this piece of code up until the posix validation may be duplicated from the run command. Consider making a small helper function for it. 

Also, can we skip variable reading when we're using `v1`? Not super important as performance isn't a big factor here so if doing it anyway keeps it simpler stick to that. Only refactor if it makes more sense that way.

Reply:
> Also, can we skip variable reading when we're using v1?

Currently, we're returning an error when template vars are supplied. This also happens when template vars are supplied via envvars. So, reading the vars should also be done in `v1` to  check whether no var is set.